File: tests/07_phases_ompss.dg/fortran/success_task_30.f90
! <testinfo>
! test_generator=config/mercurium-ompss
! </testinfo>
PROGRAM TEST
    IMPLICIT NONE

    TYPE T
        INTEGER :: Q(4)
    END TYPE T

    TYPE(T), ALLOCATABLE :: V(:)

    INTEGER :: INDEX1, UBOUND1
    REAL*8 :: RESULTS


    ALLOCATE(V(2))

    INDEX1 = 1
    UBOUND1= 2

    !$OMP TASK FIRSTPRIVATE(INDEX1, UBOUND1) INOUT(V(INDEX1) % Q(1:UBOUND1))
    !$OMP END TASK

    !$OMP TASKWAIT

    DEALLOCATE(V)
END PROGRAM TEST

File: tests/07_phases_ompss.dg/c/success_task_30.c
/*
<testinfo>
test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
test_nolink=yes
</testinfo>
*/
enum { N = 10 };

int main(int argc, char *argv[])
{
    int v[N];
    int *p = v;

    #pragma omp task inout([N]p)
    {}

    return 0;
}
